### ISSUE TITLE
webview_flutter flutter block the accessibility Z gesture in iOS

### ISSUE THREAD

**Issue by tamminivenkat** [flutter/flutter#87031]
webview_flutter flutter block the accessibility two finger Z gesture in iOS. I'm using the WebView in part of the screen. After loading subview WebView then Z gesture not working.

**Comment by darshankawar** [comment]
@tamminivenkat 
Same as https://github.com/flutter/flutter/issues/87030#issuecomment-886635232

**Comment by tamminivenkat** [comment]
Here, I can't upload my sample app of this issue. I tried attach my zip sample through dragging, But I can't.

**Comment by darshankawar** [comment]
It seems to be similar / related to the one I linked in your earlier issue, https://github.com/flutter/flutter/issues/87030. Check and confirm.

**Comment by tamminivenkat** [comment]
[lib.zip](https://github.com/flutter/flutter/files/6891838/lib.zip)
Here is the sample application files to find this bug on this WebView screen.

**Comment by darshankawar** [comment]
@tamminivenkat 
This seems to be related / similar to https://github.com/flutter/flutter/issues/30804 wherein the overall issue is related to accessibility in webview isn't working properly for iOS platform. I suggest you to follow-up in the original issue for further updates. Once that issue is fixed, this and your other issue https://github.com/flutter/flutter/issues/87030 should be fixed as well.

**Comment by darshankawar** [comment]
Also the original issue was fixed `Confirmed that the issue is gone in iOS 13.2.3 and above`. What iOS version and flutter version are you on ?

**Comment by tamminivenkat** [comment]
I am on iOS version is 14.6 and flutter version is 2.2.3

**Comment by darshankawar** [comment]
@tamminivenkat
I verified your code on latest master (2.5.0-2.0.pre.66) using iPhone 6s (OS 14.4.1). after tapping on `https:google.com`, it opened google home page, then I turned ON accessibility and used 2 fingers to scroll which led to voiceover go over all the elements on the screen one by one till the end of the screen. Below is one such screenshot which focussed on `google app` element. Likewise, it traversed to each and every element properly after using 2 finger swipe / scroll.

<img width="350" alt="Screenshot 2021-07-29 at 11 50 05 AM" src="https://user-images.githubusercontent.com/67046386/127441814-d9966220-7946-4ced-8621-e9df94bd19fe.png">


Is this a correct interpretation of your issue ? If not, provide step by step details to replicate.

**Comment by tamminivenkat** [comment]
@darshankawar, 

Use Two finger gesture not for scroll,  It's for Z'scrub gesture like bellow video. While accessibility cases, User can able to drag the screen as Z letter shape on the screen. This Z gesture scrub action back button action. At this time in native iOS 'accessibilityPerformEscape' method will be called. Please follow the bellow YouTube link.

https://www.youtube.com/watch?v=gYTMA5SEzNU 


In the above sample code, we can able get back to child screen to parent screen with Z scrub gesture. This will happens only without webview screen. With webview, Z scrub not working.

**Comment by darshankawar** [comment]
@tamminivenkat 
Thanks for the details.



> In the above sample code, we can able get back to child screen to parent screen with Z scrub gesture. This will happens only without webview screen. With webview, Z scrub not working.

I tried a sample navigation [example](https://flutter.dev/docs/cookbook/navigation/navigation-basics#interactive-example) and tried Z gesture which didn't go back to previous screen.

Can you provide a Flutter example in which the Z gesture works ?

**Comment by tamminivenkat** [comment]
In my sample app you can try without webview(Just comment the webview widget) then Z gesture will be works.

Or 

I edited my code you can check here:


[lib.zip](https://github.com/flutter/flutter/files/6906365/lib.zip)

**Comment by darshankawar** [comment]
@tamminivenkat 
Tried your updated code sample and performed Z gesture on child screen, but couldn't get it to work to go back to previous screen. See below video. You obviously won't be able to see the Z gestures I tried on white screen, but there I tried but it didn't direct me to previous screen.


https://user-images.githubusercontent.com/67046386/127641591-deb04d83-059a-45f5-afb2-e64cc02f1425.MP4

It'll be great if you could show me a working video from your end.

**Comment by tamminivenkat** [comment]
Here is the working video:

https://user-images.githubusercontent.com/75079805/127992534-0dc44823-a7e0-4412-94c6-ecd3fff723e3.mp4

**Comment by darshankawar** [comment]
@tamminivenkat 
How were you able to replicate it ? Seems it has accessibility ON ? Is it with Flutter or non-Flutter ?

**Comment by tamminivenkat** [comment]
Yes, It has accessibility ON cases.
 
and It is with Flutter.

**Comment by darshankawar** [comment]
What's the difference between the one [you](https://github.com/flutter/flutter/issues/87031#issuecomment-891689740) shared and the one when I tried  [earlier](https://github.com/flutter/flutter/issues/87031#issuecomment-889807215) ?

**Comment by tamminivenkat** [comment]
Did you enable voice over?. It seems, not enabled

**Comment by darshankawar** [comment]
Ok, I was able to replicate the issue using webview and voiceover ON, ie, the Z gesture on the webview doesn't take user back to previous screen, but I am also not sure if this should be the intended behavior. When you are opening any link using webview, the link superseeds the widget and thus may not recognize the gestures done over the platform-view ie webview.

Does the same behavior work with native iOS app ?

**Comment by tamminivenkat** [comment]
In native iOS have webview delegate methods, We can recognize and control that any links.

**Comment by darshankawar** [comment]
I see the same behavior as mentioned earlier https://github.com/flutter/flutter/issues/87031#issuecomment-892398443

Keeping this open for further investigation.

<details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 2.2.3, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.2.3 at /Users/dhs/documents/fluttersdk/flutter
    • Framework revision f4abaa0735 (5 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 92.0.4515.107

• No issues found!



```

</details>

**Comment by jmagman** [comment]
Updated code sample for webview_flutter 4.7.0
```dart
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() => runApp(const App());

class App extends StatelessWidget {
  const App({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Web Views',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  static final links = ['https://google.com'];

  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        key: const Key('123'),
        body: SafeArea(
            child: SingleChildScrollView(
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.center,
                  crossAxisAlignment: CrossAxisAlignment.stretch,
                  children: links.map((link) => _urlButton(link, context)).toList(),
                ))));
  }

  Widget _urlButton(String url, BuildContext context) {
    return Container(
        padding: const EdgeInsets.all(20.0),
        child: ElevatedButton(
          child: Text(url),
          onPressed: () => _handleURLButtonPress(url, context),
        ));
  }

  void _handleURLButtonPress(String url, BuildContext context) {
    Navigator.push(context, SlideRightRoute(page: WebViewExample(url)));
  }
}

class SlideRightRoute extends PageRouteBuilder {
  final Widget page;
  SlideRightRoute({required this.page})
      : super(
    pageBuilder: (
        BuildContext context,
        Animation<double> animation,
        Animation<double> secondaryAnimation,
        ) =>
    page,
    transitionsBuilder: (
        BuildContext context,
        Animation<double> animation,
        Animation<double> secondaryAnimation,
        Widget child,
        ) =>
    child,
    barrierDismissible: true,
  );
}

class WebViewExample extends StatefulWidget {
  final String url;
  const WebViewExample(this.url, {super.key});

  @override
  State<WebViewExample> createState() => _WebViewExampleState(url);
}

class _WebViewExampleState extends State<WebViewExample> {
  late final WebViewController controller;
  String url;
  _WebViewExampleState(this.url);

  @override
  void initState() {
    super.initState();

    controller = WebViewController()
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..setBackgroundColor(const Color(0x00000000))
      ..setNavigationDelegate(
        NavigationDelegate(
          onProgress: (int progress) {
            // Update loading bar.
          },
          onPageStarted: (String url) {},
          onPageFinished: (String url) {},
          onWebResourceError: (WebResourceError error) {},
          onNavigationRequest: (NavigationRequest request) {
            if (request.url.startsWith('https://www.youtube.com/')) {
              return NavigationDecision.prevent;
            }
            return NavigationDecision.navigate;
          },
        ),
      )
      ..loadRequest(Uri.parse(url));
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: WebViewWidget(controller: controller),
    );
  }
}
```

Confirmed this does not reproduce on https://github.com/flutter/engine/pull/51241.  See also https://github.com/flutter/flutter/issues/74246. Closing as fixed.
